src/js/calculate.js

```diff
@@ -90,46 +91,53 @@ export default {
             // time this event fires.
             //
-            // We change the order by first evaluating the non-formcontrol setvalue directives (in document order), and then
+            // We change the order by first evaluating the non-formcontrol setvalue/setgeopoint directives (in document order), and then
             // the ones with form controls.
             // https://github.com/OpenClinica/enketo-express-oc/issues/355#issuecomment-725640823
-            nodes = [ ...this.form.view.html.querySelectorAll( `.setvalue [data-setvalue][data-event*="${event.type}"]` ) ].concat(
-                this.form.filterRadioCheckSiblings( [ ...this.form.view.html.querySelectorAll( `.question [data-setvalue][data-event*="${event.type}"]` ) ] ) );
+            return [ ...this.form.view.html.querySelectorAll( `.${action} [data-${action}][data-event*="${event.type}"]` ) ].concat(
+                this.form.filterRadioCheckSiblings( [ ...this.form.view.html.querySelectorAll( `.question [data-${action}][data-event*="${event.type}"]` ) ] ) );
         } else if ( event.type === new events.NewRepeat().type ) {
             // Only this event requires specific index targeting through the "updated" object
             // We change the order by first evaluating the non-formcontrol setvalue directives (in document order), and then
             // the ones with form controls.
             // https://github.com/OpenClinica/enketo-express-oc/issues/355#issuecomment-725640823
             // https://github.com/OpenClinica/enketo-express-oc/issues/419
-            nodes = this.form.getRelatedNodes( 'data-setvalue', `.setvalue [data-event*="${event.type}"]`, event.detail ).get().concat(
-                this.form.getRelatedNodes( 'data-setvalue', `.question [data-event*="${event.type}"]`, event.detail ).get() );
+            return this.form.getRelatedNodes( `data-${action}`, `.${action} [data-event*="${event.type}"]`, event.detail ).get().concat(
+                this.form.getRelatedNodes( `data-${action}`, `.question [data-event*="${event.type}"]`, event.detail ).get() );
 
         } else if ( event.type === new events.XFormsValueChanged().type ) {
             const question = event.target.closest( '.question' );
-            nodes = question ? [ ...question.querySelectorAll( `[data-setvalue][data-event*="${event.type}"]` ) ] : nodes;
+
+            return question ? [ ...question.querySelectorAll( `[data-${action}][data-event*="${event.type}"]` ) ] : [];
+        }
+    },
+
+    /**
+     * Runs <setvalue> actions.
+     *
+     * @param {'setvalue' | 'setgeopoint'} type - the nodes relevant to the action.
+     * @param {CustomEvent} [event] - the event type that triggered the setvalue action.
+     */
+    performAction( type, event ) {
+        if ( !event ) {
+            return;
         }
 
+        if ( !this.form ) {
+            throw new Error( 'Setvalue module not correctly instantiated with form property.' );
```

Comment:
"Setvalue module" -> "Action"? I'm not really clear on what "module" refers to here.

Reply:
Module is the whole file (an ES module). It's a weird check perhaps and I think there is some historic reason for it, probably related to testing. I won't work like a regular ES module unless it's added like this: https://github.com/enketo/enketo-core/blob/c35aaba086876ff0b2588270f2fad5ed71f16c25/src/js/form.js#L199

May be worth revisiting the Form.addModule approach in the future.